EDPF.cpp

```diff
@@ -65,47 +63,31 @@ void EDPF::validateEdgeSegments()
 
 void EDPF::ComputePrewitt3x3(std::vector<short> &img)
 {
-  img.clear();
   img.resize(width * height, 0);
-
-  std::vector<int> grads(MAX_GRAD_VALUE, 0);
-
-  for (int i = 1; i < height - 1; i++)
-  {
-    for (int j = 1; j < width - 1; j++)
-    {
-      // Prewitt Operator in horizontal and vertical direction
-      // A B C
-      // D x E
-      // F G H
-      // gx = (C-A) + (E-D) + (H-F)
-      // gy = (F-A) + (G-B) + (H-C)
-      //
-      // To make this faster:
-      // com1 = (H-A)
-      // com2 = (C-F)
-      // Then: gx = com1 + com2 + (E-D) = (H-A) + (C-F) + (E-D) = (C-A) + (E-D) + (H-F)
-      //       gy = com1 - com2 + (G-B) = (H-A) - (C-F) + (G-B) = (F-A) + (G-B) + (H-C)
-      //
-      int com1 = srcImg[(i + 1) * width + j + 1] - srcImg[(i - 1) * width + j - 1];
-      int com2 = srcImg[(i - 1) * width + j + 1] - srcImg[(i + 1) * width + j - 1];
-
-      int gx = abs(com1 + com2 + (srcImg[i * width + j + 1] - srcImg[i * width + j - 1]));
-      int gy = abs(com1 - com2 + (srcImg[(i + 1) * width + j] - srcImg[(i - 1) * width + j]));
-
-      int g = gx + gy;
-
-      img[i * width + j] = g;
-      grads[g]++;
-    }  // end-for
-  }    // end-for
+  auto image = cv::Mat_(height, width, &img[0]);
```

Comment:
Is this definition meaningful even if there's `image =` at L73?

Reply:
Good catch. Updated to directly set the grad image to `gradImage` variable.